### PR TITLE
Add first set of test cases, fix various typos

### PR DESCRIPTION
**Pull request by algo26-matthias**

### CODE REVIEW

**Comment by glensc** [comment]
i'd rather rebase than merge (merge master into dev), this makes history cleaner, especially if looking at PR itself.

File: src/NamePrepData.php [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 namespace Algo26\IdnaConvert;
 
+/**
+ * @codeCoverageIgnore character mapss
```
Comment (glensc) [inline src/NamePrepData.php]: `s/mapss/maps/` typo?

File: tests/autoload.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+function test_auto_loader($className)
```
Comment (glensc) [inline tests/autoload.php]: why not just use composer autoloader? use "autoload-dev" section

File: tests/integration/IdnaConvertDecodeTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<?php
+namespace Algo26\IdnaConvert\test;
```
Comment (glensc) [inline tests/integration/IdnaConvertDecodeTest.php]: consistency? use `Test` as namespace, not lowercase

File: tests/phpunit.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/6.5/phpunit.xsd"
+         bootstrap="./autoload.php"
```
Comment (glensc) [inline tests/phpunit.xml]: same here, use `vendor/autoload.php` from composer

File: tests/autoload.php (hunk 2 above)
Comment (glensc) [reply tests/autoload.php]: example: https://github.com/eventum/rpc/blob/v4.3.0/composer.json#L21-L25

File: src/NamePrepData.php (hunk 1 above)
Comment (algo26-matthias) [reply src/NamePrepData.php]: Sure, thanks!

File: tests/autoload.php (hunk 2 above)
Comment (algo26-matthias) [reply tests/autoload.php]: Less code is always better. I will look into it and change it.

File: tests/integration/IdnaConvertDecodeTest.php (hunk 3 above)
Comment (algo26-matthias) [reply tests/integration/IdnaConvertDecodeTest.php]: You're right!

File: tests/phpunit.xml (hunk 4 above)
Comment (algo26-matthias) [reply tests/phpunit.xml]: Sure.

**Pull request merged**